Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Threading.Tasks;
+using Promact.Trappist.DomainModel.Models.TestConduct;
+using Promact.Trappist.DomainModel.DbContext;
+using Microsoft.EntityFrameworkCore;
+using System;
+
+namespace Promact.Trappist.Repository.TestConduct
+{
+    public class TestConductRepository : ITestConductRepository
+    {
+        #region Private Variables
+        #region Dependencies
+        private readonly TrappistDbContext _dbContext;
+        #endregion
+        #endregion
+
+        #region Constructor
+        public TestConductRepository(TrappistDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+        #endregion
+
+        #region Public Method
+        public async Task<bool> RegisterTestAttendeesAsync(TestAttendees model, string magicString)
+        {
+            if (await IsTestAttendeeExistAsync(model, magicString))
+            {
+                await _dbContext.TestAttendees.AddAsync(model);
+                await _dbContext.SaveChangesAsync();
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// This method used for check test attendee already exist for this test.
+        /// </summary>
+        /// <param name="model">This model object contain test attendee credential which are first name,last name,email,roll number,contact number</param>
+        /// <param name="magicString">This parameter contain test link</param>
+        /// <returns>If test attendee exist then return false else return true.</returns>
+        private async Task<bool> IsTestAttendeeExistAsync(TestAttendees model, string magicString)
+        {
+            try
+            {
+                model.TestId = (await _dbContext.Test.FirstOrDefaultAsync(x => (x.Link == magicString))).Id;
```

Comment:
again, there are chances to occur null reference exception. Please rectify this.